File: Deepseek-Coder-33b-Instruct/stage2/data_construct.c
#include <stdio.h>
#include <stdlib.h>
#include <openacc.h>

#define N 10

int main() {
    int *host_array = (int*)malloc(N * sizeof(int));
    int *device_array = (int*)malloc(N * sizeof(int));
    int i;

    // Initialize host array
    for (i = 0; i < N; i++) {
        host_array[i] = i;
    }

    // Copy host array to device
    #pragma acc data copyin(host_array[0:N]) copyout(device_array[0:N])
    {
        // Copy host array to device
        #pragma acc update device(host_array[0:N])

        // Modify device array
        #pragma acc parallel loop
        for (i = 0; i < N; i++) {
            device_array[i] = host_array[i] * 2;
        }

        // Copy device array back to host
        #pragma acc update self(device_array[0:N])
    }

    // Check if device array is correct
    for (i = 0; i < N; i++) {
        if (device_array[i] != host_array[i] * 2) {
            printf("Error: device_array[%d] = %d, expected %d\n", i, device_array[i], host_array[i] * 2);
            return 1;
        }
    }

    printf("Data construct test passed\n");
    return 0;
}

File: Deepseek-Coder-33b-Instruct/stage2/data_construct.f90
!T1:data construct,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        REAL(KIND=4), DIMENSION(N) :: a, b
        INTEGER :: i

        ! Initialize arrays
        DO i = 1, N
          a(i) = i * 1.0
          b(i) = 0.0
        END DO

        !$acc data copyin(a), copyout(b)
        !$acc parallel loop
        DO i = 1, N
          b(i) = a(i)
        END DO
        !$acc end data

        ! Check the results
        DO i = 1, N
          IF (b(i) .ne. i * 1.0) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM